resolver/resolver.cc

```diff
@@ -1282,8 +1289,11 @@ class ResolveTypeMembersAndFieldsWalk {
         typecase(
             expr, [&](const ast::Literal &a) { result = a.value; },
             [&](const ast::Cast &cast) {
-                if (cast.type != todoType() && cast.cast != core::Names::let() &&
-                    cast.cast != core::Names::uncheckedLet()) {
+                if (cast.type == core::Types::todo()) {
+                    return;
```

Comment:
I believe this is not quite the same as before unless you set `result = cast.type` before the early return

Reply:
You're correct. In this case it returns `nullptr` which is an expected return value for "we failed to extract the type".

For example, see this callsite:

```c++
            if (auto resultType = resolveConstantType(ctx, asgn->rhs)) {
                return resultType;
            } else {
                return core::Types::todo();
            }
```

If `resolveConstantType` returns `todo()`, the caller returns `todo()`. If it's `nullptr`... the caller returns `todo()`.